File: documentation-examples-scala/src/scala/dev/typr/foundations/docs/postgresql/NetworkTypes.scala
package dev.typr.foundations.docs.postgresql

import dev.typr.foundations.{PgType, PgTypes}
import dev.typr.foundations.data.{Cidr, Inet}

@SuppressWarnings(Array("unused"))
object NetworkTypes:
  //start
  val inetType: PgType[Inet] = PgTypes.inet
  val cidrType: PgType[Cidr] = PgTypes.cidr

  val addr: Inet = new Inet("192.168.1.1/24")
  //stop

File: documentation-examples-java/src/java/dev/typr/foundations/docs/postgresql/NetworkTypes.java
package dev.typr.foundations.docs.postgresql;

import dev.typr.foundations.PgType;
import dev.typr.foundations.PgTypes;
import dev.typr.foundations.data.Cidr;
import dev.typr.foundations.data.Inet;

@SuppressWarnings("unused")
public class NetworkTypes {
  // start
  PgType<Inet> inetType = PgTypes.inet;
  PgType<Cidr> cidrType = PgTypes.cidr;

  Inet addr = new Inet("192.168.1.1/24");
  // stop
}
